pramen-py/src/pramen_py/metastore/reader.py

```diff
@@ -116,19 +115,13 @@ def get_latest(
             f"{table.info_date_settings.column}={latest_date}",
         )
         logger.debug(f"Path to the latest partition is: {path}")
-        if table.format is TableFormat.parquet:
-            df = self.spark.read.parquet(path).withColumn(
-                table.info_date_settings.column,
-                F.lit(latest_date).cast(T.DateType()),
-            )
-        elif table.format is TableFormat.delta:
-            raise NotImplementedError
-        else:
-            raise NotImplementedError
-        logger.info(
-            f"Table {table_name} with the latest partition {latest_date} "
-            f"loaded"
-        )
+
+        df = self.spark.read.format(table.format.value) \
+            .load(pathlib.Path(path).as_posix()) \
+            .withColumn(table.info_date_settings.column,
```

Comment:
When using Delta format you can't read a partition directly. You need to read the base folder and then do the filtering.

If all tests pass currently, please add a failing test first before fixing this.

Reply:
Checked - the behavior is the same. Adding tests are not needed.